### ISSUE TITLE
API for random access output

### ISSUE THREAD

**Issue by thejoshwolfe** [thejoshwolfe/yazl#20]
Currently, output is provided by the client piping the `outputStream` field to the destination (file write stream, socket, etc.). This means yazl can't go back and correct local file headers once we know the file sizes and crc32's. Unfortunately, Mac's Archive Utility is so nonconformant that adding #6 ZIP64 support will break all compatibility with Mac's Archive Utility without correcting local file headers using random access output file creation. This is because of 4 bad decisions in the design of Mac's Archive Utility:

1. Mac's Archive Utility cares about the redundant information in the local file headers instead of just the authoritative information in the central directory.
2. Mac's Archive Utility requires the presence of the optional Data Descriptor section (see #7 ) sometimes.
3. Mac's Archive Utility requires that there be no dead space between entries where the optional Data Descriptor would go if Mac's Archive Utility thought that it was appropriate for it to be there.
4. Mac's Archive Utility doesn't recognize ZIP64 format for local file headers with unknown sizes and crc32. Instead, you get confusing and unpredictable behavior (not even an error message) when trying to extract the zip file.

All of these are in violation of the spec. 1-3 are simply stupid design, and I don't know what kind of logic would motivate anyone to write software like that (probably the same logic that designed the zip file spec in the first place). Number 4 above is the most directly relevant problem to this issue, and it isn't quite due to stupid design; it's just oversights and sloppy design.

So basically, Mac's Archive Utility is the Internet Explorer of zip file extractors; it is the notepad.exe of zip file extractors. Horribly designed, nonconformant, unhelpful pain in my ass.

That being ranted, I will probably not have motivation to implement this for a while.

**Comment by thejoshwolfe** [comment]
It's not quite this hopeless. see #6.

**Comment by thejoshwolfe** [comment]
I now believe most of what I claimed in the OP for this issue is wrong. I suspect I did some bad science when I was doing preliminary testing for this issue. My current findings are that we don't need this feature.
